lib/index.ts

```diff
@@ -522,6 +543,14 @@ export function emit(rootDecl: TopLevelDeclaration, rootFlags = ContextFlags.Non
         newline();
     }
 
+    function writeFunctionType(f: FunctionDeclaration) {

```

Comment:
This should get refactored such that `writeFunctionType` is accepting a `FunctionType` as it doesn't make sense to emit a declaration inline.